app/src/main/java/com/android/joinme/ui/chat/ChatScreen.kt

```diff
@@ -0,0 +1,498 @@
+ * with existing ProfilePhotoImage component for consistency across the app.
+ *
+ * @param photoUrl URL of the user's profile photo, or null for default avatar
+ * @param userName Name of the user for accessibility descriptions
+ * @param modifier Modifier for positioning (typically Alignment.Bottom in message row)
+ */
+@Composable
+private fun UserAvatar(photoUrl: String?, userName: String, modifier: Modifier = Modifier) {
+  ProfilePhotoImage(
+      photoUrl = photoUrl,
+      contentDescription = "$userName's avatar",
+      modifier = modifier,
+      size = Dimens.IconSize.medium,
+      shape = CircleShape,
+      showLoadingIndicator = false // Don't show spinner for small avatars in chat
+      )
+}
+
+/**
+ * Message input field with send button.
+ *
+ * @param text Current input text
+ * @param onTextChange Callback when text changes
+ * @param onSendClick Callback when send button is clicked
+ * @param sendButtonColor The color for the send button
+ */
+@Composable
+private fun MessageInput(
+    text: String,
+    onTextChange: (String) -> Unit,
+    onSendClick: () -> Unit,
+    sendButtonColor: androidx.compose.ui.graphics.Color
+) {
+  Surface(shadowElevation = Dimens.Elevation.small, color = MaterialTheme.colorScheme.surface) {
+    Row(
+        modifier =
+            Modifier.fillMaxWidth()
+                .padding(horizontal = Dimens.Padding.medium, vertical = Dimens.Padding.small),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(Dimens.Spacing.small)) {
+          OutlinedTextField(
+              value = text,
+              onValueChange = onTextChange,
+              modifier = Modifier.weight(1f).testTag(ChatScreenTestTags.MESSAGE_INPUT),
+              placeholder = {
+                Text(
+                    text = "Type your message here...",
+                    color = MaterialTheme.colorScheme.onSurfaceVariant)
+              },
+              shape = RoundedCornerShape(Dimens.CornerRadius.pill),
+              colors = MaterialTheme.customColors.outlinedTextField(),
+              maxLines = 4)
+
+          // Send button
+          IconButton(
+              onClick = onSendClick,
+              enabled = text.isNotBlank(),
+              modifier =
+                  Modifier.size(Dimens.TouchTarget.minimum)
+                      .background(
+                          color =
+                              if (text.isNotBlank()) sendButtonColor
+                              else MaterialTheme.colorScheme.surfaceVariant,
+                          shape = CircleShape)
+                      .testTag(ChatScreenTestTags.SEND_BUTTON)) {
+                Icon(
+                    imageVector = Icons.AutoMirrored.Filled.Send,
+                    contentDescription = "Send message",
+                    tint = MaterialTheme.customColors.buttonContentColor)
```

Comment:
Question:

Visually, what do you think of this change, I think it's better, but tell me what do you think ?

```suggestion
          // Send button
          IconButton(
              onClick = onSendClick,
              enabled = text.isNotBlank(),
              modifier = Modifier.size(Dimens.TouchTarget.minimum)
                      .background(
                          color = MaterialTheme.colorScheme.surfaceVariant
                          shape = CircleShape)
                      .testTag(ChatScreenTestTags.SEND_BUTTON)) {
                Icon(
                    imageVector = Icons.AutoMirrored.Filled.Send,
                    contentDescription = "Send message button",
                    tint = if (text.isNotBlank()) {
                                  sendButtonColor
                              } else {
                                  MaterialTheme.colorScheme.onSurfaceVariant
                              }
                    )
```

Reply:
I really appreciate your effort in proposing an alternative. However, your proposal removes visual prominence and introduces color contrast violations. Instead, I’m going with:
```
// Send button
IconButton(
    onClick = onSendClick,
    enabled = text.isNotBlank(),
    modifier =
        Modifier.size(Dimens.TouchTarget.minimum)
            .background(
                color =
                    if (text.isNotBlank()) sendButtonColor
                    else MaterialTheme.colorScheme.surfaceVariant,
                shape = CircleShape)
            .testTag(ChatScreenTestTags.SEND_BUTTON)) {
      Icon(
          imageVector = Icons.AutoMirrored.Filled.Send,
          contentDescription = "Send message button",
          tint =
              if (text.isNotBlank()) onSendButtonColor
              else MaterialTheme.colorScheme.onSurfaceVariant)
```
resulting in the following:
<img width="295" height="56" alt="Screenshot 2025-11-15 at 11 05 32 PM" src="https://github.com/user-attachments/assets/010d6c6d-1274-42ad-9d15-dc0bfa55a925" />
<img width="294" height="56" alt="Screenshot 2025-11-15 at 11 05 48 PM" src="https://github.com/user-attachments/assets/9c96ab05-0e19-4b3b-8865-7a2100847912" />
<img width="295" height="56" alt="Screenshot 2025-11-15 at 11 06 00 PM" src="https://github.com/user-attachments/assets/c7dd8efa-8c4e-4779-90b1-b29a3fe2eb26" />
<img width="294" height="52" alt="Screenshot 2025-11-15 at 11 06 10 PM" src="https://github.com/user-attachments/assets/25163d22-adb3-4d50-907d-4e36dffdf367" />